app/Enrollment.php

```diff
@@ -4,19 +4,48 @@
 
 use Illuminate\Database\Eloquent\Model;
 use Carbon\Carbon;
+use Illuminate\Support\Facades\DB;
 
 class Enrollment extends Model
 {
-    public function class() {
-        return $this->belongsTo(Classes::class, 'classes_id');
-    }
+  public function enrollable()
+  {
+    return $this->morphTo();
+  }
 
-    public function date() {
-        return $this->belongsTo(ClassDate::class, 'class_dates_id');
-    }
+  public function student() {
+    return $this->belongsTo(Student::class, 'student_id');
+  }
 
-    public function friendlyDate($date)
-    {
-        return (Carbon::parse($date)->format('D\, M\. jS Y'));
+  public function friendlyDate($date)
+  {
+    if (isset($date)) {
+      return (Carbon::parse($date)->format('D\, M\. jS Y'));
+    } else {
+      return null;
     }
+      
+  }
+
+  public function otherEnrollmentsInClass(Enrollment $enrollment)
+  {
+    if ($enrollment->enrollable->class->season->SeasonType == 1) {
+      $otherEnrollments = DB::table('class_days')
```

Comment:
I don't like this. Try to either do something with Eloquent, or wrap all of this logic up into 1 function in the model and call it.